server/routes/users.js

```diff
@@ -2,14 +2,272 @@
+router.get('/:user_id/events', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT event_id FROM event_checkin WHERE user_id = $1 UNION SELECT event_id FROM event_registration WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all events a user is hosting
+router.get('/:user_id/host', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT event_id FROM event_host WHERE host_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all the user's majors
+router.get('/:user_id/majors', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT major_id FROM user_major WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all positions a user is seeking
+router.get('/:user_id/positions', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT position_id FROM user_position WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all the user's occupations
+router.get('/:user_id/occupations', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT occupation_id FROM user_occupation WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+
+//Get the user's diet information
+router.get('/:user_id/diet', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT diet_id FROM user_diet WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get a user's favorite events
+router.get('/:user_id/favorite', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT event_id FROM favorite_events WHERE user_id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get a user's favorite events
+//HELP NEEDED
+/*
+router.get('/search', function(req, res, next) {
+  const name = req.query.name;
+  db.query('SELECT id FROM swe_user WHERE CONCAT(first_name, ' ', last_name) ILIKE '%$1%'', [name], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+*/
+
+//Get all users whose emails contain the substring email
+//HELP NEEDED
+router.get('/search', function(req, res, next) {
+  const email = req.query.email;
+  db.query('SELECT id FROM swe_user WHERE email ILIKE '%$1%'', [email], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
 });
 
+
+router.get('/filter', function(req, res, next) {
```

Comment:
This route is probably one of the more difficult ones. Glad to see you combined all the options into one route, but the logic should be changed. Luckily, with the users API the filters are all independent (afaik), so the filters should have an AND relationship with each other when fetching results from the database. Think on this and how best to do this without too much redundancy, but don't overthink it. It just involves modifying <10 lines